assets/src/components/detours/detourMap.tsx

```diff
@@ -254,3 +271,32 @@ const OriginalRouteShape = ({
     )}
   </>
 )
+
+interface DivertedRouteShapeProps extends PropsWithChildren {
+  segments: RouteSegments
+}
+
+const DivertedRouteShape = ({
+  segments: { beforeDetour, detour, afterDetour },
+}: DivertedRouteShapeProps) => (
+  <>
+    <Polyline
+      weight={6}
+      interactive={false}
+      positions={beforeDetour.map(shapePointToLatLngLiteral)}
```

Comment:
nit(non-blocking): I can't pull this down and check right now, but since there's no referential identity preserved here, do the polylines continuously rerender and have the zoom issue we saw the other time? We may need to memo these if so.

Reply:
I didn't notice any issue when I was playing around with this, but.... maybe this is like two bugs that cancel each other out?

Reply:
@firestack I'm not seeing a user-facing issue, but intuitively it feels like this is _wrong_.

Do you think it's worth it to `useMemo` here anyway, just to be on the safe side? (And to avoid making people's browsers do extra work?)

Reply:
the `useMemo` docs say to only use it as a performance optimization, so if we're not seeing issues, then I think we're fine? Easy to do later on if it's an issue